packages/purpose-process/src/services/purposeService.ts

```diff
@@ -903,3 +1011,26 @@ const updatePurposeInternal = async (
     ),
   };
 };
+
+const getDailyCallsFromPurposeToClone = (purposeToClone: Purpose): number => {
+  const nonWaitingVersions = purposeToClone.versions.filter(
+    (v) => v.state !== purposeVersionState.waitingForApproval
+  );
+
+  const versionsToSearch =
+    nonWaitingVersions.length > 0
+      ? nonWaitingVersions
+      : purposeToClone.versions;
+
+  const sortedVersions = versionsToSearch.slice().sort((v1, v2) => {
+    if (v1.createdAt > v2.createdAt) {
+      return -1;
+    } else if (v1.createdAt < v2.createdAt) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+
+  return sortedVersions.length > 0 ? sortedVersions[0].dailyCalls : 0;
```

Comment:
```suggestion
  const sortedVersions = [...versionsToSearch].sort(
    (v1, v2) => v2.createdAt.getTime() - v1.createdAt.getTime()
  );
  return sortedVersions[0]?.dailyCalls ?? 0;
```
It can be simplified like this

Reply:
I updated the sorting, thanks! For the last line I prefer the ternary operator.